src/lib/snyk/dependencies.ts

```diff
@@ -7,31 +7,56 @@ import { IssueWithPaths } from '../types'
 const displayDependenciesChangeDetails = async (snykDepsJsonResults: any, monitoredProjectDepGraph: any, packageManager: string, newVulns: IssueWithPaths[], newLicenseIssues: IssueWithPaths[]) => {
     let snykTestGraph: depgraph.DepGraph
     if(snykDepsJsonResults && snykDepsJsonResults.depGraph){ // Getting graph
+        
         snykTestGraph = depgraph.createFromJSON(snykDepsJsonResults.depGraph)
     } else { // Getting legacy dep tree
         snykTestGraph = await depgraph.legacy.depTreeToGraph(snykDepsJsonResults as depgraph.legacy.DepTree, packageManager)
     }
     
     //const snykTestDeps = snykTestGraph.getPkgs()
     const snykTestDirectDepsNodeIDs = snykTestGraph.toJSON().graph.nodes[0].deps.map(dep => dep.nodeId)
-    const snykTestDirectDeps = snykTestGraph.toJSON().pkgs.filter(pkg => snykTestDirectDepsNodeIDs.indexOf(pkg.id) > 0)
-    const snykTestIndirectDeps = snykTestGraph.toJSON().pkgs.filter(pkg => snykTestDirectDepsNodeIDs.indexOf(pkg.id) <= 0)
+    const snykTestDirectDepsPkgIDs: Array<string> = []
+    
+        snykTestDirectDepsNodeIDs.forEach(nodeId => {
+            const node = snykTestGraph.toJSON().graph.nodes.find(node => node.nodeId == nodeId)
+            if(!node){
+                throw new Error(`Could not find node in graph ${nodeId}`)
+            } else {
+                snykTestDirectDepsPkgIDs.push(node.pkgId)
+            }
+            
+        })
+    
     
+
+    const snykTestDirectDeps = snykTestGraph.toJSON().pkgs.filter(pkg => snykTestDirectDepsPkgIDs.indexOf(pkg.id) > 0)
+    const snykTestIndirectDeps = snykTestGraph.toJSON().pkgs.filter(pkg => snykTestDirectDepsPkgIDs.indexOf(pkg.id) <= 0)
     
     const snykProjectGraph = depgraph.createFromJSON(monitoredProjectDepGraph.depGraph as depgraph.DepGraphData)
-    //const snykProjectDeps = snykTestGraph.getPkgs()
+
     const snykProjectDirectDepsNodeIDs = snykProjectGraph.toJSON().graph.nodes[0].deps.map(dep => dep.nodeId)
```

Comment:
do we need to do this again, there is `snykTestDirectDepsNodeIDs` already?

Reply:
yes, here it's the direct dep nodeIds from the monitored snapshot coming via the API.